### PR TITLE
Implement Channel Proposal: Send, Sub/Unsub Notification and Respond sequence.

### PR DESCRIPTION
**Pull request by manoranjith**
#### Description

Implement Channel Proposal: Send, Sub/Unsub Notification and Respond sequence.

##### Category

Implementation Task 

##### Relevant issue

Relates to #83.

#### Testing

Happy path test using integration setup include.

##### Steps to run the tests

1. Start the ganache-cli node with this command:
```
ganache-cli -b 1 --account="0x1fedd636dbc7e8d41a0622a2040b86fea8842cef9d4aa4c582aad00465b7acff,100000000000000000000" --account="0xb0309c60b4622d3071fad3e16c2ce4d0b1e7758316c187754f4dd0cfb44ceb33,100000000000000000000"
``` 
2. Run the test in session package:
```
cd session && go test -tags=integration -count=1 -cover -coverprofile=c.out -run Test_Integ_Role -v
```
3.  

#### Checklist 

- [x] Name is added to the NOTICE file, if it is not present already.
- [x] Changes are rebased onto the target branch.

### CODE REVIEW

File: session/sessiontest/config.go [hunk 1]
```diff
@@ -58,10 +58,10 @@ func NewConfigFile(t *testing.T, config session.Config) string {
 // A contacts file is created with the given set of peers and path to it is added in the config.
 // This function also registers cleanup functions for removing all the temp files and dirs after the test.
 //
-// This function uses its own prng seed instead of receiving it from the calling test function because the
-// specific seed is required to match the accounts funded in on the blockchain.
-func NewConfig(t *testing.T, contacts ...perun.Peer) session.Config {
-	rng := rand.New(rand.NewSource(1729))
+// This function returnes session config with user on-chain addresses that are funded on blockchcian when using a
```
Comment (anagha-ks) [inline session/sessiontest/config.go]:
```suggestion
// This function returns session config with user on-chain addresses that are funded on blockchain when using a
```

File: session/sessiontest/config.go [hunk 2]
```diff
@@ -58,10 +58,10 @@ func NewConfigFile(t *testing.T, config session.Config) string {
...
+// a particular seed for prng. In this case, the first two consequitive calls to this function will return
```
Comment (anagha-ks) [inline session/sessiontest/config.go]:
```suggestion
// particular seed for prng. In this case, the first two consecutive calls to this function will return
```

**Review by anagha-ks — CHANGES_REQUESTED** [review]

File: session/sessiontest/config.go (hunk 1 above)
Comment (manoranjith) [reply session/sessiontest/config.go]: Updated.

File: session/sessiontest/config.go (hunk 2 above)
Comment (manoranjith) [reply session/sessiontest/config.go]: Updated.

**Review by anagha-ks — APPROVED** [review]

**Pull request merged**